eZ/Bundle/EzPublishSolrSearchEngineBundle/DependencyInjection/Configuration.php

```diff
@@ -43,15 +125,131 @@ protected function addConnectionsSection( ArrayNodeDefinition $node )
                 ->info( "Name of the default connection" )
             ->end()
             ->arrayNode( "connections" )
-                ->info( "Solr Search Engine connections configuration" )
-                ->useAttributeAsKey( "connection_name" )
-                ->performNoDeepMerging()
-                ->prototype( "array" )
-                    ->children()
-                        ->scalarNode( "server" )
-                            ->isRequired()
-                            ->info( "Address of the Solr server" )
-                            ->example( "https://username:password@hostname.com/path:1234" )
+            ->info( "Solr Search Engine connection configuration" )
+            ->useAttributeAsKey( "connection_name" )
+            ->performNoDeepMerging()
+            ->prototype( "array" )
+                ->beforeNormalization()
+                    ->ifTrue(
+                        function( $v )
+                        {
+                            return (
+                                empty( $v["entry_points"]["content"] ) &&
+                                !empty( $v["cluster"]["content"] )
+                            );
+                        }
+                    )
+                    ->then(
+                        function( $v )
+                        {
+                            // If Content search entry points are not provided use cluster endpoints
+                            $v["entry_points"]["content"] = array_values( $v["cluster"]["content"] );
+                            return $v;
+                        }
+                    )
+                ->end()
+                ->beforeNormalization()
+                    ->ifTrue(
+                        function( $v )
+                        {
+                            return (
+                                empty( $v["entry_points"]["location"] ) &&
+                                !empty( $v["cluster"]["location"] )
+                            );
+                        }
+                    )
+                    ->then(
+                        function( $v )
+                        {
+                            // If Location search entry points are not provided use cluster endpoints
+                            $v["entry_points"]["location"] = array_values( $v["cluster"]["location"] );
+                            return $v;
+                        }
+                    )
+                ->end()
+                ->children()
+                    ->arrayNode( "entry_points" )
```

Comment:
Because this defines which endpoints are used as entry points. For example, having `http://localhost:1000/solr/core0` as the entry point would mean all requests are made towards it, and search would be distributed in a `shards` parameter, like:

```
http://localhost:1000/solr/core0?shards=something.com:8983/solr/core1,other.com:1234/solr/core2
```